engine/src/main/java/org/pentaho/di/core/util/serialization/JAXBMetaSerializer.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2018 by Hitachi Vantara : http://www.pentaho.com
+ *
+ * **************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.pentaho.di.core.util.serialization;
+
+import com.google.common.base.Throwables;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+import javax.xml.XMLConstants;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.StringWriter;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.nio.charset.Charset.defaultCharset;
+
+/**
+ * Converts StepMetaProps to/from an XML string using JAXB.
+ */
+public class JAXBMetaSerializer {
```

Comment:
Would it be better to name this class something more generic like `MetaSerializer`? Under the covers it uses JAXB but do I as a user of class care? JAXB could be swapped out to use some other implementation and to me it wouldn't matter since it serializes my Meta.

Reply:
You _may_ care.  I think the nice thing about having a serializable form like MetaProps is that you could serialize it with different strategies-- a JSON serializer, or binary, or whatever.  If you
're using the JAXB serializer that has implications for the XML generated for storage.

Reply:
I guess I could suggest XMLMetaSerializer.

Reply:
👍